src/MIDI.hpp

```diff
@@ -1397,51 +1375,20 @@ inline void MidiInterface<Transport, Settings, Platform>::UpdateLastSentTime()
 // - Channel messages are passed to the output whether their channel
 //   is matching the input channel and the filter setting
 template<class Transport, class Settings, class Platform>
-void MidiInterface<Transport, Settings, Platform>::thruFilter(Channel inChannel)
+void MidiInterface<Transport, Settings, Platform>::thruFilter()
 {
-    // If the feature is disabled, don't do anything.
-    if (!mThruActivated || (mThruFilterMode == Thru::Off))
-        return;
+   if (!mThruFilterCallback(mMessage))
+      return;
+
+   MidiMessage thruMessage = mThruMapCallback(mMessage);
 
     // First, check if the received message is Channel
     if (mMessage.type >= NoteOff && mMessage.type <= PitchBend)
```

Comment:
We should check the type of the mapped message rather than the received one (same below), to handle for example a NoteOn -> SysEx conversion.

Reply:
It didn't occur to me that someone might send an entirely different message type, but of course.

Reply:
I had a use-case once where I wanted to use a keyboard as a preset loader for a synth, and mapped each received note to a specific SysEx patch.